core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1193,8 +1193,10 @@ class DAGScheduler(
         // TODO: This will be really slow if we keep accumulating shuffle map stages
         for ((shuffleId, stage) <- shuffleToMapStage) {
           stage.removeOutputsOnExecutor(execId)
-          val locs = stage.outputLocs.map(list => if (list.isEmpty) null else list.head).toArray
-          mapOutputTracker.registerMapOutputs(shuffleId, locs, changeEpoch = true)
+          if (!runningStages.contains(stage)) {
```

Comment:
The additional test passes without this change.  Is this necessary for correctness?  If so, we need a better test for it.  If its not for correctness, and just an optimization, I'm a little hesitant about including it, I'd to better understand why it would help.